### PR TITLE
Transition to the fdcan-rs crate

### PR DESCRIPTION
**Pull request by luctius**
As discussed in #34, the FDCAN part has been transitioned to a new crate. This PR uses that crate instead of our own implementation.

### CODE REVIEW

**Comment by kevswims** [comment]
@luctius is there anything I can do to help get this in? I have a board with an STM32g4 and 2 CAN transceivers on it that I can do testing on if needed. At the moment I am using a super hacky driver in my own code to get the CAN working but I would love to move to this if possible.

**Comment by luctius** [comment]
I've updated this to fdcan 0.1.2. It builds; but if you have time I would mind if you could check that everything still works.

File: Cargo.toml [hunk 1]
```diff
@@ -23,6 +23,14 @@ static_assertions = "1.1"
 version = "0.7.7"
 features = ["critical-section-single-core"]
 
+[dependencies.fdcan]
+# git = "https://github.com/stm32-rs/fdcan.git"
```
Comment (kevswims) [inline Cargo.toml:27]: Probably don't want to merge with these commented bits in here.

File: src/lib.rs [hunk 2]
```diff
@@ -62,8 +62,8 @@ pub use crate::stm32::interrupt;
 
 pub mod adc;
 pub mod bb;
-// pub mod can;
-pub mod fdcan;
+pub mod can;
+// pub mod fdcan;
```
Comment (kevswims) [inline src/lib.rs:66]: Remove commented code?

**Review by kevswims — CHANGES_REQUESTED** [review]
I made some changes to the example itself to get this going on my hardware and I was able to get the the two channels on my board to talk to each other. Functionally, I think this is good to go in.

I can put up a follow-on PR that cleans up the example to allow running it either on physical hardware or in loopback mode for testing. I should be able to get that up in a day or two.

**Pull request merged**